src/emulator/types.ts

```diff
@@ -133,7 +152,50 @@ export class EmulatorLog {
   }
 
   log(): void {
```

Comment:
I'd love to see a comment here or somewhere that explains (to our best knowledge) why we need what appears to be an overly complex manual buffering system.

Reply:
Can we modify the stdout buffer size: https://stackoverflow.com/questions/23429499/stdout-buffer-issue-using-node-child-process

Reply:
It says `Default: 1024 * 1024.` which is definitely not what I was observing.  I think that's the max total buffered data that has not been flushed and I was running into the 8192 per-write limit